### ISSUE TITLE
add support for vue script setup

### ISSUE THREAD

**Issue by zynth17** [fullcalendar/fullcalendar-vue#170]
i'm using vue 3.2.20 and the latest vite 2.6.14. 
i'm using the new script setup.
i got an error when importing the fullcalendar package
here is the error : 
![image](https://user-images.githubusercontent.com/45350572/142174572-889301e0-3866-4ab7-966d-c75fb3264322.png)

and this error affect all my other component which is pretty bad,  all my other custom components are showing an error like this.
if im not importing fullcalendar on my project , it goes back to normal.
![image](https://user-images.githubusercontent.com/45350572/142174799-85b8a6fd-0520-4303-a0bb-258a3bd6b54a.png)

here is my reproduction test. you can see the error when you run npm run build on the terminal.  but with vscode you can already see the error without building the project.
https://stackblitz.com/edit/vitejs-vite-nkdmzt?file=src/App.vue

**Comment by acerix** [comment]
If I understand, you mean adding support for "script setup" like on this page?

https://v3.vuejs.org/api/sfc-script-setup.html

**Comment by zynth17** [comment]
> If I understand, you mean adding support for "script setup" like on this page?
> 
> 
> 
> https://v3.vuejs.org/api/sfc-script-setup.html

yes correct.

**Comment by DanjBethel** [comment]
Anything on this? Having the exact same issue.

**Comment by zynth17** [comment]
> Anything on this? Having the exact same issue.

nope, i ended up creating my own calendar.

**Comment by DanjBethel** [comment]
> 

Well that royally sucks. Sigh

**Comment by DanjBethel** [comment]
@acerix Having this issue as well. Really would like to use this

**Comment by MedLabs** [comment]
it's still incompatible with vite 2 script setup :(
got to find another calendar library

**Comment by hongquan** [comment]
Is the error message seen only in IDE or also in `yarn vite build`?

If you don't get error with `yarn vite build`, it is just IDE problem. If you are using VS Code + Volar extension, it is Volar problem and the [workaround](https://github.com/fullcalendar/fullcalendar-vue/issues/173#issuecomment-1007559246) is to downgrade Volar.

**Comment by DanjBethel** [comment]
> Is the error message seen only in IDE or also in `yarn vite build`?
> 
> If you don't get error with `yarn vite build`, it is just IDE problem. If you are using VS Code + Volar extension, it is Volar problem and the [workaround](https://github.com/fullcalendar/fullcalendar-vue/issues/173#issuecomment-1007559246) is to downgrade Volar.

Yeah. It is an IDE problem. It builds fine. I downgraded as the article suggested a while ago but it still didn't work. I regrettably had to deactivate Volar for this project as I desperately wanted to use this calendar.

**Comment by zynth17** [comment]
> Maybe your VS Code silently upgrade Volar again?

for me the problem come from fullcalendar itself.
there is literally no problem at all from the volar side because this only happen if i import fullcalendar library.
and downgrading volar is not an option because need to use the latest typescript version.

**Comment by hongquan** [comment]
Another ugly workaround is https://github.com/johnsoncodehk/volar/discussions/592#discussioncomment-2033669 and now I can upgrade Volar back :D

**Comment by IceBlizz6** [comment]
Would be great if this could be prioritized.
Our project is using Vue CLI instead of Vite and it doesn't seem like the above solution works for us.
In either case i believe this is a critical issue that should be prioritized regardless of existing hacks and workarounds.

**Comment by hongquan** [comment]
The root cause here is that FullCalendar depends on Preact for its virtual DOM. But Preact has some side-effect related to JSX that pollutes Volar. This problem happens if your project use some 3rd-party libraries which depend on Preact/React, not just FullCalendar. So I think this is more a challenge for Volar than FullCalendar issue, because if FullCalendar not depend on Preact, other of your dependencies will do.

**Comment by baboon-king** [comment]
> The root cause here is that FullCalendar depends on Preact for its virtual DOM. But Preact has some side-effect related to JSX that pollutes Volar. This problem happens if your project use some 3rd-party libraries which depend on Preact/React, not just FullCalendar. So I think this is more a challenge for Volar than FullCalendar issue, because if FullCalendar not depend on Preact, other of your dependencies will do.

you right. 
maybe we can overwrite `namespace` for Preact/React's JSX

**Comment by baboon-king** [comment]
I overwrited path for `preact/compat`

template worked in `.vue`
```jsonc
{
  "extends": "@vue/tsconfig/tsconfig.web.json",
  "include": ["env.d.ts", "src/**/*", "src/**/*.vue"],
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"],
      "preact/compat": ["anything.d.ts"] // fix JSX's type bugs https://github.com/fullcalendar/fullcalendar-vue/issues/170
    },
    "types": ["element-plus/global"]
  },

  "references": [
    {
      "path": "./tsconfig.config.json"
    }
  ]
}
```
![image](https://user-images.githubusercontent.com/63645337/179462517-09b0b3b2-afe7-4edd-9823-da622b35a558.png)

**Comment by RihanArfan** [comment]
Here's a code example of script setup + composition API with FullCalendar. https://github.com/fullcalendar/fullcalendar-vue/issues/172#issuecomment-1201270026

Make sure to do what other people are suggesting about preact, whether it be deleting the types in the node module with a postinstall script or overriding it with tsconfig otherwise Volar will throw a fit.
